fluent-syntax/src/ftlstream.js

```diff
@@ -93,20 +93,36 @@ export class FTLParserStream extends ParserStream {
     return undefined;
   }
 
-  isIDStart() {
-    if (this.ch === undefined) {
+  isCharIDStart(ch) {
+    if (ch === undefined) {
       return false;
     }
 
-    const cc = this.ch.charCodeAt(0);
-    return ((cc >= 97 && cc <= 122) || // a-z
-            (cc >= 65 && cc <= 90) || // A-Z
-             cc === 95); // _
+    const cc = ch.charCodeAt(0);
+    return (cc >= 97 && cc <= 122) || // a-z
+           (cc >= 65 && cc <= 90); // A-Z
+  }
+
+  isMessageIDStart() {
+    if (this.currentIs('-')) {
+      this.peek();
+    }
+
+    const ch = this.currentPeek();
+    const isID = this.isCharIDStart(ch);
+    this.resetPeek();
+    return isID;
   }
 
   isNumberStart() {
-    const cc = this.ch.charCodeAt(0);
-    return ((cc >= 48 && cc <= 57) || cc === 45); // 0-9
+    if (this.currentIs('-')) {
+      this.peek();
+    }
+
+    const cc = this.currentPeek().charCodeAt(0);
+    const isDigit = cc >= 48 && cc <= 57; // 0-9
+    this.resetPeek();
+    return isDigit;
```

Comment:
what's the value of this change?

Reply:
It makes `isNumberStart` consistent with `isMessageIDStart` and more importantly, it considers the char after the leading `-`. The previous version would return true upon seeing a dash right away and more checking was required in `getLiteral`.